test/integrated/full_land.jl

```diff
@@ -78,6 +78,56 @@ for FT in (Float32, Float64)
         @test model.canopy == canopy
         @test model.snow == snow
     end
+
+    @testset "LandModel with no soilco2 model constructor, FT=$FT" begin
+        toml_dict = LP.create_toml_dict(FT)
+        domain = Domains.global_domain(FT)
+        atmos, radiation = ClimaLand.prescribed_analytic_forcing(FT; toml_dict)
+        forcing = (; atmos, radiation)
+        prognostic_land_components = (:canopy, :snow, :soil)
+
+        # Soil model
+        soil = Soil.EnergyHydrology{FT}(
+            domain,
+            forcing,
+            toml_dict;
+            prognostic_land_components,
+            additional_sources = (ClimaLand.RootExtraction{FT}(),),
+        )
+
+        # Canopy model
+        surface_domain = Domains.obtain_surface_domain(domain)
+        LAI = TimeVaryingInput((t) -> FT(1.0))
+        ground = ClimaLand.PrognosticGroundConditions{FT}()
+        canopy_forcing = (; atmos, radiation, ground)
+        canopy = Canopy.CanopyModel{FT}(
+            surface_domain,
+            canopy_forcing,
+            LAI,
+            toml_dict;
+            prognostic_land_components,
+        )
+
+        # Snow model
+        dt = FT(180)
+        snow = SnowModel(
+            FT,
+            surface_domain,
+            forcing,
+            toml_dict,
+            dt;
+            prognostic_land_components,
+        )
+        soilco2 = nothing
+        model = LandModel{FT}(canopy, snow, soil, soilco2)
+
+        # The constructor has many asserts that check the model
+        # components, so we don't need to check them again here.
+        @test model.soil == soil
+        @test isnothing(model.soilco2)
+        @test model.canopy == canopy
+        @test model.snow == snow
```

Comment:
we could also do `Y, p, cds = initialize(model)` and make sure Y and p dont have soilco2 as a key? and check `land_components(model)`